src/Extensions/Logging/Internal/EventSource/OmexLogEventSender.cs

```diff
@@ -0,0 +1,182 @@
+			string tagId = string.IsNullOrEmpty(eventId.Name) ? TagIdAsString(eventId.Id) : eventId.Id.ToString(CultureInfo.InvariantCulture);
+			string traceIdAsString = traceId.ToHexString();
+
+			//Event methods should have all information as parameters so we are passing them each time
+			// Posible Breaking changes:
+			// 1. CorrelationId type changed from Guid ?? Guid.Empty
+			// 2. TransactionId type Changed from uint ?? 0u
+			// 3. ThreadId type Changed from string
+			// 4. TagName to events so it should be also send
+			switch (level)
+			{
+				case LogLevel.None:
+					break;
+				case LogLevel.Trace:
+					m_eventSource.LogSpamServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Spam", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Debug:
+					m_eventSource.LogVerboseServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Verbose", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Information:
+					m_eventSource.LogInfoServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Info", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Warning:
+					m_eventSource.LogWarningServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Warning", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Error:
+				case LogLevel.Critical:
+					m_eventSource.LogErrorServiceMessage(applicationName, serviceName, machineId, buildVersion, s_processName, partitionId, replicaId, activityId, traceIdAsString, "Error", category, tagId, eventId.Name, threadId, message);
+					break;
+				default:
+					throw new ArgumentException("Unknown EventLevel: " + level);
+			}
+		}
+
+
+		public bool IsEnabled(LogLevel level) =>
+		level switch
+		{
+			LogLevel.None => false,
+			_ => m_eventSource.IsEnabled()
+		};
+
+
+		public bool IsReplayableMessage(LogLevel logLevel) =>
+			logLevel switch
+			{
+				LogLevel.Trace => true,
+				LogLevel.Debug => true,
+				_ => false
+			};
+
+
+		public void ReplayLogs(Activity activity)
+		{
+			if (activity is ReplayableActivity replayableActivity)
+			{
+				foreach (LogMessageInformation log in replayableActivity.GetLogEvents())
+				{
+					LogMessage(activity.Id, activity.TraceId, log.Category, LogLevel.Information, log.EventId, log.ThreadId, log.Message);
+				}
+			}
+		}
+
+
+		private readonly OmexLogEventSource m_eventSource;
+		private readonly IServiceContext m_serviceContext;
+		private readonly IMachineInformation m_machineInformation;
+		private static readonly string s_processName;
```

Comment:
Can this be `const`?

Reply:
I don't think it can since it's evaluated in the constructor